File: easy/letterCasePct/letterCasePct.rb
File.open(ARGV[0]).each_line do |line|
	line = line.chomp
	upper = 0
	line.each_char{|c|
		c.ord > 64 && c.ord < 91 && upper += 1	
	}
	upperPct = upper.to_f * 100.0 / line.length.to_f
	lowerPct = 100.0 - upperPct
	puts "lowercase: %0.2f " % [lowerPct] + "uppercase: %0.2f"  % [upperPct]
end

File: easy/letterCasePct/letterCasePct.cs
using System;
using System.IO;
class program
{
    static void Main(string[] args)
    {
        //using (StreamReader sr = new StreamReader("test_file.txt"))
        using (StreamReader sr = File.OpenText(args[0]))
        {
            string line;
            while ((line = sr.ReadLine()) != null)
            {
                char[] letters = line.ToCharArray();

                decimal total = letters.Length;
                decimal upper = 0;

                foreach(char letter in letters)
                {
                    if(Char.IsUpper(letter))
                    {
                        upper++;
                    }
                }

                decimal upperpct = upper * 100 / total;

                upperpct = Math.Round(upperpct, 2);

                decimal lowerpct = 100 - upperpct;


                Console.WriteLine("lowercase: {0:0.00} uppercase: {1:0.00}", lowerpct, upperpct);

            }
        }

        Console.Read();
    }
}

